lib/gooddata/lcm/actions/collect_data_product.rb

```diff
@@ -45,7 +45,14 @@ def call(params)
             params.gdc_logger.info "Using data product 'default' since none was specified in brick parameters"
             data_product_id = 'default'
           end
-          data_product = domain.data_products(data_product_id)
+
+          begin
+            data_product = domain.data_products(data_product_id)
+          rescue RestClient::BadRequest
+            params.gdc_logger.info "Can not find DataProduct #{params.data_product}, creating it instead"
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)